src/YoutubeAtomPlayer.tsx

```diff
@@ -0,0 +1,276 @@
+import React, { useEffect, useRef } from 'react';
+import YouTubePlayer from 'youtube-player';
+
+import type { AdTargeting, ImageSource, VideoEventKey } from './types';
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import {
+    AdsConfig,
+    buildAdsConfigWithConsent,
+    disabledAds,
+} from '@guardian/commercial-core';
+import { log } from '@guardian/libs';
+
+type Props = {
+    assetId: string;
+    overrideImage?: ImageSource[];
+    posterImage?: ImageSource[];
+    adTargeting?: AdTargeting;
+    consentState?: ConsentState;
+    height: number;
+    width: number;
+    title?: string;
+    origin?: string;
+    eventEmitters: ((event: VideoEventKey) => void)[];
+    autoPlay: boolean;
+    loadPlayer: boolean;
+    setPlayerReady: (ready: true) => void;
+};
+
+declare global {
+    interface Window {
+        onYouTubeIframeAPIReady: unknown;
+    }
+}
+
+type YoutubeCallback = (e: YT.PlayerEvent & YT.OnStateChangeEvent) => void;
+
+/**
+ * youtube-player doesn't have a type definition
+ * Based on https://github.com/gajus/youtube-player
+ */
+type YoutubePlayerType = {
+    on: (state: string, callback: YoutubeCallback) => YoutubeCallback;
+    off: (callback: YoutubeCallback) => void;
+    loadVideoById: (videoId: string) => void;
+    playVideo: () => void;
+    getCurrentTime: () => number;
+    getDuration: () => number;
+    getPlayerState: () => number;
+};
+
+type ProgressEvents = {
+    hasSentPlayEvent: boolean;
+    hasSent25Event: boolean;
+    hasSent50Event: boolean;
+    hasSent75Event: boolean;
+};
+
+/**
+ * ProgressEvents are a ref, see below
+ */
+const createOnStateChangeListener = (
+    progressEvents: ProgressEvents,
+    eventEmitters: Props['eventEmitters'],
+): YT.PlayerEventHandler<YT.OnStateChangeEvent> => (event) => {
+    const loggerFrom = 'YoutubeAtomPlayer onStateChange';
+    log('dotcom', {
+        from: loggerFrom,
+        event,
+    });
+
+    // event.target is the actual underlying YT player
+    const player = event.target;
+
+    if (event.data === YT.PlayerState.PLAYING) {
+        if (!progressEvents.hasSentPlayEvent) {
+            log('dotcom', {
+                from: loggerFrom,
+                msg: 'start play',
+                event,
+            });
+            eventEmitters.forEach((eventEmitter) => eventEmitter('play'));
+            progressEvents.hasSentPlayEvent = true;
+
+            setTimeout(() => {
```

Comment:
I know this is copied code but we're using `window.setTimeout` below so maybe we can do the same here?